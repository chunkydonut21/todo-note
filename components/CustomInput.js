import { useTheme } from '@react-navigation/native'
import { useField } from 'formik'
import { useState } from 'react'
import { StyleSheet, Text, TextInput, View } from 'react-native'
import Colours from '../Colours'

export const CustomInput = ({ name, label, ...props }) => {
  const [field, meta, helpers] = useField(name)
  const [focused, setFocused] = useState(false)
  const { colors } = useTheme()

  return (
    <View style={styles.container}>
      <Text style={[styles.label, { color: colors.text }]}>{label}</Text>
      <TextInput
        style={styles.input}
        {...props}
        onChangeText={helpers.setValue}
        onBlur={() => {
          helpers.setTouched(true)
          setFocused(false)
        }}
        onFocus={() => setFocused(true)}
        value={field.value}
      />
      {meta.touched && meta.error ? <Text style={{ color: colors.notification }}>{meta.error}</Text> : <Text> </Text>}
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    marginBottom: 8
  },
  label: {
    fontSize: 16
  },
  input: {
    borderRadius: 28,
    color: Colours.black,
    paddingHorizontal: 10,
    width: '100%',
    backgroundColor: '#fff',
    marginVertical: 10,
    height: 48,
    borderWidth: StyleSheet.hairlineWidth,
    paddingHorizontal: 8
  }
})
