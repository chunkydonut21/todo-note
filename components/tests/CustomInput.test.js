import { render } from '@testing-library/react-native'
import { useField } from 'formik'
import React from 'react'
import { CustomInput } from '../CustomInput'

jest.mock('formik')

describe('Testing custom input field that we use in Authentication flow', () => {
  it('renders the custom input field', () => {
    useField.mockReturnValue([
      { value: '' },
      { error: '', touched: false },
      { setValue: jest.fn(), setTouched: jest.fn() }
    ])
    const tree = render(<CustomInput name="sampleField" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('shows validation error when touched', () => {
    useField.mockReturnValue([
      { value: 'Test value' },
      { error: 'This field is required', touched: true },
      { setValue: jest.fn(), setTouched: jest.fn() }
    ])
    const { getByText } = render(<CustomInput name="test" />)
    expect(getByText('This field is required')).toBeTruthy()
  })

  it('shows text label field correctly', () => {
    useField.mockReturnValue([
      { value: 'Test value' },
      { error: '', touched: false },
      { setValue: jest.fn(), setTouched: jest.fn() }
    ])
    const tree = render(<CustomInput name="sampleField" label="Test label" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
