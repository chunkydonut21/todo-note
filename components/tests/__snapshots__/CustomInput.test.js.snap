// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing custom input field that we use in Authentication flow renders the custom input field 1`] = `
<View
  style={
    {
      "marginBottom": 8,
    }
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
        },
        {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
  />
  <TextInput
    onBlur={[Function]}
    onChangeText={[MockFunction]}
    onFocus={[Function]}
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 28,
        "borderWidth": 0.5,
        "color": "#212121",
        "height": 48,
        "marginVertical": 10,
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
    value=""
  />
  <Text>
     
  </Text>
</View>
`;

exports[`Testing custom input field that we use in Authentication flow shows text label field correctly 1`] = `
<View
  style={
    {
      "marginBottom": 8,
    }
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
        },
        {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
  >
    Test label
  </Text>
  <TextInput
    onBlur={[Function]}
    onChangeText={[MockFunction]}
    onFocus={[Function]}
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 28,
        "borderWidth": 0.5,
        "color": "#212121",
        "height": 48,
        "marginVertical": 10,
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
    value="Test value"
  />
  <Text>
     
  </Text>
</View>
`;

exports[`TextInput renders the custom input field 1`] = `
<View
  style={
    {
      "marginBottom": 8,
    }
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
        },
        {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
  />
  <TextInput
    onBlur={[Function]}
    onChangeText={[MockFunction]}
    onFocus={[Function]}
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 28,
        "borderWidth": 0.5,
        "color": "#212121",
        "height": 48,
        "marginVertical": 10,
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
    value=""
  />
  <Text>
     
  </Text>
</View>
`;
